Sources/CodeEditTextView/EmphasisManager/EmphasisManager.swift

```diff
@@ -0,0 +1,362 @@
+            return nil
+        }
+
+        var bounds = shapePath.bounds
+        bounds.origin.y += 1 // Move down by 1 pixel
+
+        // Create text layer
+        let textLayer = CATextLayer()
+        textLayer.frame = bounds
+        textLayer.backgroundColor = NSColor.clear.cgColor
+        textLayer.contentsScale = textView.window?.screen?.backingScaleFactor ?? 2.0
+        textLayer.allowsFontSubpixelQuantization = true
+        textLayer.zPosition = 2
+
+        // Get the font from the attributed string
+        if let font = originalString.attribute(.font, at: 0, effectiveRange: nil) as? NSFont {
+            textLayer.font = font
+        } else {
+            textLayer.font = NSFont.systemFont(ofSize: NSFont.systemFontSize)
+        }
+
+        updateTextLayer(textLayer, with: originalString, emphasis: emphasis)
+        return textLayer
+    }
+
+    private func updateTextLayer(
+        _ textLayer: CATextLayer,
+        with originalString: NSAttributedString,
+        emphasis: Emphasis
+    ) {
+        let text = NSMutableAttributedString(attributedString: originalString)
+        text.addAttribute(
+            .foregroundColor,
+            value: emphasis.inactive ? getInactiveTextColor() : NSColor.black,
+            range: NSRange(location: 0, length: text.length)
+        )
+        textLayer.string = text
+    }
+
+    private func getInactiveTextColor() -> NSColor {
+        if textView?.effectiveAppearance.name == .darkAqua {
+            return .white
+        }
+        return .black
+    }
+
+    private func applyPopAnimation(to layer: CALayer) {
+        let scaleAnimation = CAKeyframeAnimation(keyPath: "transform.scale")
+        scaleAnimation.values = [1.0, 1.25, 1.0]
+        scaleAnimation.keyTimes = [0, 0.3, 1]
+        scaleAnimation.duration = 0.1
+        scaleAnimation.timingFunctions = [CAMediaTimingFunction(name: .easeOut)]
+
+        layer.add(scaleAnimation, forKey: "popAnimation")
+    }
+
+    private func applyFadeOutAnimation(to layer: CALayer, textLayer: CATextLayer?) {
+        let fadeAnimation = CABasicAnimation(keyPath: "opacity")
+        fadeAnimation.fromValue = 1.0
+        fadeAnimation.toValue = 0.0
+        fadeAnimation.duration = 0.1
+        fadeAnimation.timingFunction = CAMediaTimingFunction(name: .easeOut)
+        fadeAnimation.fillMode = .forwards
+        fadeAnimation.isRemovedOnCompletion = false
+
+        layer.add(fadeAnimation, forKey: "fadeOutAnimation")
+
+        if let textLayer = textLayer {
+            if let textFadeAnimation = fadeAnimation.copy() as? CABasicAnimation {
+                textLayer.add(textFadeAnimation, forKey: "fadeOutAnimation")
+            }
+        }
+
+        // Remove both layers after animation completes
+        DispatchQueue.main.asyncAfter(deadline: .now() + fadeAnimation.duration) {
+            layer.removeFromSuperlayer()
```

Comment:
```suggestion
        DispatchQueue.main.asyncAfter(deadline: .now() + fadeAnimation.duration) { [weak layer, weak textLayer] in
            layer?.removeFromSuperlayer()
```

We may have removed these layers before this completes, so they should be weakly referenced to avoid a reference cycle.